src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,158 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
+        if (key == null) {
+            putForNullKey(value);
+            return;
+        }
+        int index = calculateBucketIndex(key);
+        Node<K, V> newNode = new Node<>(key, value, null);
+        if (table[index] == null) {
+            table[index] = newNode;
+        } else {
+            Node<K, V> currentNode = table[index];
+            while (currentNode != null) {
+                if (Objects.equals(currentNode.key, key)) {
+                    currentNode.value = value;
+                    return;
+                }
+                if (currentNode.next == null) {
+                    break;
+                }
+                currentNode = currentNode.next;
+            }
+            currentNode.next = newNode;
+        }
+        size++;
+        if (size > table.length * DEFAULT_LOAD_FACTOR) {
+            resize();
+        }
+    }
 
+    private void putForNullKey(V value) {
+        if (table[0] == null) {
+            table[0] = new Node<>(null, value, null);
+            size++;
+        } else {
+            Node<K, V> currentNode = table[0];
+            while (currentNode.next != null) {
+                if (currentNode.key == null) {
+                    currentNode.value = value;
+                    return;
+                }
+                currentNode = currentNode.next;
+            }
+            if (currentNode.key == null) {
+                currentNode.value = value;
+            } else {
+                currentNode.next = new Node<>(null, value, null);
+                size++;
+            }
+            if (size > table.length * DEFAULT_LOAD_FACTOR) {
```

Comment:
I think you may move this if check to the resize method (it will take less space this way)